http/src/main/java/io/micronaut/http/converters/HttpConverterRegistrar.java

```diff
@@ -36,14 +36,14 @@
 @Prototype
 public class HttpConverterRegistrar implements TypeConverterRegistrar {
 
-    private final Provider<ResourceResolver> resourceResolver;
+    private final BeanProvider<ResourceResolver> resourceResolver;
 
     /**
      * Default constructor.
      *
      * @param resourceResolver The resource resolver
      */
-    protected HttpConverterRegistrar(Provider<ResourceResolver> resourceResolver) {
+    protected HttpConverterRegistrar(BeanProvider<ResourceResolver> resourceResolver) {
```

Comment:
is this internal? If not need to keep the constructor, deprecate the old one and add `@Inject` to this one

Reply:
Corrected. Those classes should be internal